### PR TITLE
add support for parsing "create domain" statement

### PR DESCRIPTION
**Pull request by cvng**
## What kind of change does this PR introduce?

Feature: try to add support for "create domain" statement as requested in #51

## What is the current behavior?

Parser panics:

```sh
could not find node for token Token { kind: Create, text: "CREATE", span: 0..6, token_type: ReservedKeyword } at depth 1
```

## What is the new behavior?

Parser returns:

```sh
  CreateDomainStmt@0..37
    Create@0..6 "CREATE"
    Whitespace@6..7 " "
    DomainP@7..13 "DOMAIN"
    Whitespace@13..14 " "
    Ident@14..28 "us_postal_code"
    Whitespace@28..29 " "
    As@29..31 "AS"
    Whitespace@31..32 " "
    TypeName@32..36
      TextP@32..36 "TEXT"
    Ascii59@36..37 ";"
```

## Additional context

Not sure of what exactly should go in the `CreateDomainStmt` branch, but I have tried with these variations and it worked:

```sql
CREATE DOMAIN name [ AS ] data_type [ constraint [ ... ] ]
```

### CODE REVIEW

File: crates/codegen/src/get_node_properties.rs
Comment (psteinroe) [inline crates/codegen/src/get_node_properties.rs:540]: `check` should be part of the `Constraint` node, right? its currently implement as

```rust
        "Constraint" => quote! {
            match n.contype {
                // ConstrNotnull
                2 => {
                    tokens.push(TokenProperty::from(Token::Not));
                    tokens.push(TokenProperty::from(Token::NullP));
                },
                // ConstrDefault
                3 => tokens.push(TokenProperty::from(Token::Default)),
                // ConstrCheck
                6 => tokens.push(TokenProperty::from(Token::Check)),
                // ConstrPrimary
                7 => {
                    tokens.push(TokenProperty::from(Token::Primary));
                    tokens.push(TokenProperty::from(Token::Key));
                },
                // ConstrForeign
                10 => tokens.push(TokenProperty::from(Token::References)),
                _ => panic!("Unknown Constraint {:#?}", n.contype),
            }
        },
```

similarly, `As` should be part of the `TypeName` node.

**Review by psteinroe — CHANGES_REQUESTED** [review]
thanks for the contribution! can you check the node output with debug logs enabled and ensure that the properties match their nodes? 

also, what would help you in figuring this out faster? I think we can improve here.

**Comment by cvng** [comment]
Got it, I have removed the extra token, thanks!

For the `As` token (which is optional), it has no effect under the `TypeName` node, so I had to keep it under the `CreateDomainStmt` node or the parser still panics with:

```sh
could not find node for token Token { kind: As, text: "AS", span: 29..31, token_type: ReservedKeyword } at depth 2
```

As a side-note: `TypeName` node starts at `location: Some(32)` so maybe this is why?
```sh
1: Node {
    kind: TypeName,
    depth: 1,
    properties: [
        TokenProperty {
            value: Some(
                "text",
            ),
            kind: None,
        },
    ],
    location: Some(
        32,
    ),
},
```

File: crates/parser/src/codegen.rs
Comment (cvng) [inline crates/parser/src/codegen.rs:100]: Here, for the range `0..32` I would expect this order:

```rust
TokenProperty::from(SyntaxKind::Create),
TokenProperty::from(SyntaxKind::DomainP),
TokenProperty::from("us_postal_code".to_string()),
TokenProperty::from(SyntaxKind::As),
```

Comment (psteinroe) [reply crates/parser/src/codegen.rs:100]: ahh sorry - the order is not relevant. I fixed the test suite in #62 to ignore the order when comparing the vectors. it should pass after you merge main into you branch. thanks!

**Review by psteinroe — APPROVED** [review]
lgtm! once you merge main tests should pass too :)

**Pull request merged**
